### PR TITLE
Group editing & deletion

### PR DESCRIPTION
**Pull request by TsarN**
For now without permission checking, still working on group membership

### CODE REVIEW

**Comment by PrabSG** [comment]
Has this branch pulled in the changes on master? There appears to be a lot of repeated changes when looking at the files changed

File: coffee-chats/src/main/java/com/google/step/coffee/Nullable.java [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package com.google.step.coffee;
+
+public @interface Nullable {}
```
Comment (PrabSG) [inline coffee-chats/src/main/java/com/google/step/coffee/Nullable.java:3]: I believe this annotation is already built into javax.annotation is it not? I am not sure what the reasoning here is to recreate it

Comment (TsarN) [reply coffee-chats/src/main/java/com/google/step/coffee/Nullable.java:3]: It doesn't unless we want to add another dependency to pom.xml

File: coffee-chats/pom.xml [hunk 2]
```diff
@@ -21,6 +21,12 @@
   </properties>
 
   <dependencies>
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>jsr305</artifactId>
+      <version>3.0.2</version>
+    </dependency>
```
Comment (PrabSG) [inline coffee-chats/pom.xml:28]: Where is this used?

**Review by PrabSG — APPROVED** [review]

File: coffee-chats/src/main/java/com/google/step/coffee/data/GroupStore.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        .setDescription("")
+        .setOwnerId(UserManager.getCurrentUserId())
+        .build());
+  }
+
+  public void delete(Key key) {
+    datastore.delete(key);
+  }
+
+  public Group put(Group group) {
```
Comment (DarthGandalf) [inline coffee-chats/src/main/java/com/google/step/coffee/data/GroupStore.java:26]: Need comments. How is group in argument different from return value?

File: coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupDeleteServlet.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import javax.servlet.annotation.WebServlet;
+import java.io.IOException;
+
+@WebServlet("/api/groupDelete")
+public class GroupDeleteServlet extends JsonServlet {
+  private GroupStore groupStore = new GroupStore();
+
+  @Override
+  public Object post(JsonServletRequest request) throws IOException, HttpError {
+    // TODO(tsarn): also check that the user has write access to the group
```
Comment (DarthGandalf) [inline coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupDeleteServlet.java:18]: please open a bug and mention it here

File: coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupInfoServlet.java [hunk 5]
```diff
@@ -4,16 +4,40 @@
...
 import java.io.IOException;
 
 @WebServlet("/api/groupInfo")
 public class GroupInfoServlet extends JsonServlet {
-  @Override
-  public Object get(JsonServletRequest request) throws IOException, HttpError {
+  private GroupStore groupStore = new GroupStore();
+
+  private Group getGroup(JsonServletRequest request) throws HttpError {
+    // TODO(tsarn): also check that the user has write access to the group
```
Comment (DarthGandalf) [inline coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupInfoServlet.java:18]: Why should getting information about the group require write access?

File: coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupDeleteServlet.java (hunk 4 above)
Comment (TsarN) [reply coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupDeleteServlet.java:18]: I have already done that todo in another branch, so it doesn't really matter

File: coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupInfoServlet.java (hunk 5 above)
Comment (TsarN) [reply coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupInfoServlet.java:18]: I have already done that todo in another branch, so it doesn't really matter

Comment (DarthGandalf) [reply coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupInfoServlet.java:18]: You didn't answer my question

File: coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupDeleteServlet.java (hunk 4 above)
Comment (DarthGandalf) [reply coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupDeleteServlet.java:18]: Okay. I haven't see PR yet

File: coffee-chats/src/main/java/com/google/step/coffee/data/GroupStore.java (hunk 3 above)
Comment (DarthGandalf) [reply coffee-chats/src/main/java/com/google/step/coffee/data/GroupStore.java:30]: Ideally, every function should have comments. Probably with an exception of functions with length of one line only

**Review by DarthGandalf — APPROVED** [review]

File: coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupInfoServlet.java (hunk 5 above)
Comment (TsarN) [reply coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupInfoServlet.java:18]: Yeah, it shouldn't, I have already fixed it already, don't know what else to say

Comment (DarthGandalf) [reply coffee-chats/src/main/java/com/google/step/coffee/servlets/GroupInfoServlet.java:18]: From your previous comment I understood that you already added this check.

How am I supposed to guess that what you did is to remove it instead? (or did you?) You didn't link to that change

**Pull request merged**